modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/modal/Modal.js

```diff
@@ -117,6 +119,12 @@ function Modal({
 			Liferay.on('destroyPortlet', cleanUpListeners);
 		}
 
+		function handleIFrameLoad() {
+			setLoading(false);
+		}
+
+		iframeRef.current?.addEventListener('load', handleIFrameLoad);
```

Comment:
I remember loading was very tricky to get right in the `frontend-js-web` version. Many use cases, custom events in DXP on open and load, React quirks with iframe, performance and accessibility concerns. I am hesitant to start poking this hornets nest in this version.

We could push changes as they are in this PR, as I would consider them a good-enough patch until we migrate to DXP Modal version. Or, we can revert back to `'modal-permissions'`, again as a temporary solution until the migation. I am fine with either one solution. Which one do you prefer?

Reply:
DXP Modal = The one in `frontend-js-web`, used in many places in DXP
FDS Modal = The one in `frontend-data-set-web`, used just in FDS